technical/indicators/indicators.py

```diff
@@ -1268,3 +1268,35 @@ def tv_hma(dataframe: DataFrame, length: int = 9, field="close") -> DataFrame:
     dataframe.drop("h", inplace=True, axis=1)
 
     return dataframe
+
+
+def trama(dataframe, length: int = 22, field="close"):
+    """
+    Name : Tradingview "Trend Regularity Adaptive Moving Average"
+    Pinescript Author : LuxAlgo
+    Link : tradingview.com/script/p8wGCPi6-Trend-Regularity-Adaptive-Moving-Average-LuxAlgo/
+
+    Args :
+        dataframe : Pandas Dataframe
+        timeperiod : TRAMA length
+        source : Field to use for the calculation
+
+    Returns :
+        dataframe : Pandas DataFrame with new columns 'trama'
+    """
+
+    import talib.abstract as ta
+
+    hh = ta.MAX(dataframe[field], length)
+    ll = ta.MIN(dataframe[field], length)
+    hhll = np.where(np.diff(hh) > 0, 1, 0) + np.where(np.diff(ll) < 0, 1, 0)
```

Comment:
I don't think this is correct, actually.

The original pinescript is making the diff first, and then does the max on top of that.

```
hh = max(sign(change(highest(length))),0)
ll = max(sign(change(lowest(length))*-1),0)
```

so maybe something around this (incomplete, untested - and switched to pandas, as to me, that transports the intend better )

```python
    change = df[field].diff()
    hh = np.maximum(np.sign(change.rolling(length).max()), 0)
    ll = np.maximum(np.sign(change.rolling(length).min()) * -1, 0)
     # ...
```

Reply:
@bkamuz the latest changes don't really go into this topic ... what's your opinion on this ?

Reply:
@bkamuz i'd apreciate if you could take a look at the above.
i think the implemented formula is wrong - as the sequence of calls deviates from pinescript.